### DISCUSSION TITLE
stuck at create-backingfiles-partition: Modifying partition table for backing files partition...

### DISCUSSION THREAD

**Discussion by mlduncombe** [marcone/teslausb#820] [Q&A] [answered]
I having problems getting things up and running.

teslausb-headless-setup-log gets stick at create-backingfiles-partition: Modifying partition table for backing files partition...

If I 

/etc/rc.local

then I see 

Downloaded /tmp/create-backingfiles-partition.sh ...
DATA_DRIVE not set. Proceeding to SD card setup
create-backingfiles-partition: Checking existing partitions...
firstbackingfilesmutable 124735488
first mutable 124530688
create-backingfiles-partition: Modifying partition table for backing files partition...
warning: /dev/mmcblk0: partition 5 is not defined yet
Checking that no-one is using this disk right now ... FAILED

This disk is currently in use - repartitioning is probably a bad idea.
Umount all file systems, and swapoff all swap partitions on this disk.
Use the --no-reread flag to suppress this check.

Disk /dev/mmcblk0: 59.5 GiB, 63864569856 bytes, 124735488 sectors
Units: sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 512 bytes / 512 bytes
Disklabel type: dos
Disk identifier: 0x8b3aede6

Old situation:

Device         Boot     Start       End   Sectors  Size Id Type
/dev/mmcblk0p1 *         8192    520191    512000  250M  c W95 FAT32 (LBA)
/dev/mmcblk0p2         520192   4714496   4194305    2G 83 Linux
/dev/mmcblk0p3        4714497 124530687 119816191 57.1G 83 Linux
/dev/mmcblk0p4      124530688 124735487    204800  100M 83 Linux

/dev/mmcblk0p5: Extended partition does not exists. Failed to add logical partition.
Leaving.

**Comment by marcone** [comment]
This:
```
/dev/mmcblk0p1 * 8192 520191 512000 250M c W95 FAT32 (LBA)
/dev/mmcblk0p2 520192 4714496 4194305 2G 83 Linux
/dev/mmcblk0p3 4714497 124530687 119816191 57.1G 83 Linux
/dev/mmcblk0p4 124530688 124735487 204800 100M 83 Linux
```
seems to show that TeslaUSB already successfully added the required partitions (the last two shown above). Not sure what 5th partition it is trying to create there. I did change something with respect to partitioning yesterday, so maybe there's an edge case I missed. I'll take a look.
What device/image are you using?

**Comment by marcone** [comment]
Found the problem. Will have a fix soon.

**Comment by marcone** [comment, marked as answer]
Should be fixed now.

**Comment by mlduncombe** [comment]
Looks like its fixed

Am now seeing 

TESLAUSB_SETUP_FINISHED 

and  get

Sun 11 Feb 08:33:45 GMT 2024 : All done.

in the setup log

Will test in car later

update: tested and working well in the car

thanks
